src/coreclr/src/jit/liveness.cpp

```diff
@@ -1980,11 +1980,10 @@ void Compiler::fgComputeLifeLIR(VARSET_TP& life, BasicBlock* block, VARSET_VALAR
                             GenTree* data =
                                 store->OperIs(GT_STOREIND) ? store->AsStoreInd()->Data() : store->AsBlk()->Data();
                             data->SetUnusedValue();
+
                             if (data->isIndir())
                             {
-                                // This is a block assignment. An indirection of the rhs is not considered
-                                // to happen until the assignment so mark it as non-faulting.
-                                data->gtFlags |= GTF_IND_NONFAULTING;
+                                data->gtType = TYP_INT;
```

Comment:
Could you please add a comment why we need to change type here?

Reply:
I'm now calling `gtTransformUnusedIndirection` here and there is a large comment in that function.